docker-compose.yml

```diff
@@ -1,11 +1,17 @@
 services:
   workspace:
-    pull_policy: "build"
-    build:
-      context: .
-    command: "pnpm exec docusaurus start --host 0.0.0.0"
+    build: .
     ports:
       - "3000:3000"
-    working_dir: "/app"
     volumes:
-      - ".:/app:cached"
+      - ./docs:/app/docs
+      - ./news:/app/news
+      - ./api_versioned_docs:/app/api_versioned_docs
+      - ./api_versioned_sidebars:/app/api_versioned_sidebars
+      - ./src:/app/src
+      - ./static:/app/static
+      - ./custom.css:/app/custom.css
+      - ./package.json:/app/package.json
+      - ./api_versions.json:/app/api_versions.json
+      - ./default-sidebars.js:/app/default-sidebars.js
+      - ./docusaurus.config.js:/app/docusaurus.config.js
```

Comment:
This seems fine, but I worry a bit that we may forget to add new important files to this. Not a blocking review comment but something I want to bring up in general.

I'm not a JS dev so forgive the possibly obtuse question: is there a way to change `node_modules` to live in another directory entirely (say `/opt/node_modules`) so that we can just sync in all of `app` and just not care?

Reply:
I'm not a JS dev either, but maybe someone can chime in / correct later. From previous research most ideas are to block `node_module` from importing by making it a separate volume.

- named volume:
  - prevents node_modules from being shared between host/container
  - but to update dependencies you need to `docker volume rm <dalamud-docs-...>` before rebuilding the image
- anon volume:
  - prevents node_modules from being shared between host/container
  - needs a `docker compose down` (to remove the association between container and volume) or `docker compose down -v` [(docs)](https://docs.docker.com/engine/reference/commandline/compose_down/#options) to skip below
  - since the volume still exists needs `docker volume prune` or `docker volume rm <anon-volume-id>`
  - Note: Like named volume but more things to keep track of
- node_modules in parent directory with empty named volume for `node_modules`
  - prevents node_modules from being shared between host/container
  - doesn't interfere with builds
  - didn't run with `pnpm` as it couldn't find node_modules; did work if starting with `npm`
  - no additional steps to update dependencies

I'm not sure what the consequences between starting with `npm start` or `pnpm start` are. I would assume that there's not much of a difference, (looks like it just calls `docusaurus <command>` (as defined in `package.json`) but am missing experience there.